test/test_optim.py

```diff
@@ -112,7 +112,15 @@ def closure():
 
             initial_value = closure().item()
             for _ in range(20):
-                loss = optimizer.step(closure)
+                if torch._dynamo.is_compiling() and not optim_info.step_requires_closure:
+                    # we disable torch.compile with the closure argument, so in order to actually
+                    # test compiling the optimizer, we separate out the closure here.
+                    # LBFGS requires the closure arg but torch.compile doesn't support LBFGS
+                    # so it's fine to run the disabled path.
+                    loss = closure()
+                    optimizer.step()
+                else:
+                    loss = optimizer.step(closure)
```

Comment:
It's fine to just flip this to
```suggestion
                if optim_info.step_requires_closure:
                    loss = optimizer.step(closure)
                else:
                    # we disable torch.compile with the closure argument, so in order to actually
                    # test compiling the optimizer, we separate out the closure here.
                    loss = closure()
                    optimizer.step()
                    
```